### PR TITLE
add docs for message hub

### PR DESCRIPTION
**Pull request by csantanapr**

### CODE REVIEW

File: docs/catalog.md [hunk 1]
```diff
@@ -627,7 +627,119 @@ The `/whisk.system/watson-speechToText/speechToText` action converts audio speec
...
+            "key": null,
+            "offset": 421761,
+            "topic": "mytopic",
+            "value": "{\"title\": \"Some string\", \"amount\": 5, \"isAwesome\": true}"
+        }
+    ]
+}
+```
+
+### Messages are Batched
```
Comment (markusthoemmes) [inline docs/catalog.md]: batched

File: docs/catalog.md [hunk 2]
```diff
@@ -627,7 +627,119 @@ The `/whisk.system/watson-speechToText/speechToText` action converts audio speec
   }
   ```
  
- 
+## Using the Message Hub package
+
+This package allows you to create triggers that react when messages are posted to [Message Hub](https://developer.ibm.com/messaging/message-hub/) service instance on Bluemix.
```
Comment (markusthoemmes) [inline docs/catalog.md]: are posted to [a] Message Hub service instance

File: docs/catalog.md [hunk 3]
```diff
@@ -627,7 +627,119 @@ The `/whisk.system/watson-speechToText/speechToText` action converts audio speec
...
+
+|Name|Type|Description|
+|---|---|---|
+|kafka_brokers_sasl|JSON Array of Strings|This parameter is an array of `<host>:<port>` strings which comprise the brokers in your Message Hub instance|
+|user|String|Your Message Hub user name|
+|password|String|Your Message Hub password|
+|topic|String|The topic you would like the trigger to listen to|
+|kafka_admin_url|URL String|The URL of the Message Hub admin REST interface|
+|api_key|String|Your Message Hub API key|
+|isJSONData|Boolean (Optional - default=false)|When set to `true` this will cause the feed to attempt the message content as JSON before passing it along as the trigger payload.|
```
Comment (markusthoemmes) [inline docs/catalog.md]: this will cause the feed to [try to parse] the message content as JSON...

**Comment by csantanapr** [comment]
@markusthoemmes thanks for the review.
Please take a look at dc556d9d121a24ba436da95c8694bc6fdc88eacd

File: docs/catalog.md [hunk 4]
```diff
@@ -627,7 +627,119 @@ The `/whisk.system/watson-speechToText/speechToText` action converts audio speec
   }
   ```
  
- 
+## Using the Message Hub package
+
+This package allows you to create triggers that react when messages are posted to a [Message Hub](https://developer.ibm.com/messaging/message-hub/) service instance on Bluemix.
+
+### Creating a Trigger that Listens to an Message Hub Instance
```
Comment (bjustin-ibm) [inline docs/catalog.md]: ...that Listens to ~an~ **a** Message Hub Instance

File: docs/catalog.md [hunk 5]
```diff
@@ -627,7 +627,119 @@ The `/whisk.system/watson-speechToText/speechToText` action converts audio speec
...
+After creating a trigger, the system will monitor the specified topic in your messaging service. When new messages are posted, the trigger will be fired.
+
+The payload of that trigger will contain a `messages` field which is an array of messages that have been posted since the last time your trigger fired. Each message object in the array will contain the following fields:
+- topic
+- partition
+- offset
+- key
+- value
+
+In Kafka terms, these fields should be self-evident. However, the `value` requires special consideration. If the `isJSONData` parameter was set `false` (or not set at all) when the trigger was created, the `value` field will be the raw value of the posted message. However, if `isJSONData` was set to `true` when the trigger was created, the system will make attempt to parse this value as a JSON object, on a best-effort basis. If parsing is successful, then the `value` in the trigger payload will be the resulting JSON object.
```
Comment (bjustin-ibm) [inline docs/catalog.md]: ... the system will ~make~ attempt to parse

**Comment by csantanapr** [comment]
@bjustin-ibm I made updates based on your feedback here 9ccff1e498968b5bdd5e55a4e2c86aa1e08fc12c

**Comment by bjustin-ibm** [comment]
Rebased/squashed/merged as 15a336e3a4343a8a37bbdba8f6c092617ea6d0d1

**Pull request closed without merging**